internal/operator-controller/authorization/rbac_test.go

```diff
@@ -241,7 +241,7 @@ func TestPreAuthorize_Failure(t *testing.T) {
 		fakeClient := setupFakeClient(limitedClusterRole)
 		preAuth := NewRBACPreAuthorizer(fakeClient)
 		missingRules, err := preAuth.PreAuthorize(context.TODO(), &exampleClusterExtension, strings.NewReader(testManifest))
-		require.Error(t, err)
+		require.NoError(t, err)
```

Comment:
but aren't these failure type tests that should require an error?

Reply:
oh, I see we're handling the nil properly now

Reply:
but then maybe this and the next test should change names and we should test for what is in them and not what's not? Name change like "preauthorize finds the missing RBAC rules"

Reply:
+1. We should change `require.NotEqual` to a `require.Equal` with the expected missing rules.

Reply:
Done